ap_ruby_sdk/lib/ap_ruby_sdk/transaction.rb

```diff
@@ -7,7 +7,6 @@ class Transaction < ApiResource
     attr_accessor :customer,
                   :customerId,
                   :payment,
-                  :token,
```

Comment:
Yes, the flow is correct.
Regarding tokens, there are Transaction token and PhoneVerification token.
You can create a transaction either with Payment object or Transaction token.
PhoneVerification token doesn't depend on how do you create a transaction (with Transaction token or Payment object).